hoprd/db/api/src/lib.rs

```diff
@@ -0,0 +1,137 @@
+//! Crate for accessing database(s) of a HOPR node.
+//! Functionality defined here is meant to be used mostly by other higher-level crates.
+
+pub mod db;
+pub mod errors;
+pub mod metadata;
+
+pub use sea_orm::DatabaseConnection;
+pub use sea_orm::DatabaseTransaction;
+
+use crate::metadata::HoprdDbMetadataOperations;
+use async_trait::async_trait;
+use futures::future::BoxFuture;
+use sea_orm::TransactionTrait;
+
+use crate::db::HoprdDb;
+use crate::errors::{DbError, Result};
+
+/// Primary key used in tables that contain only a single row.
+pub const SINGULAR_TABLE_FIXED_ID: i32 = 1;
+
+/// Shorthand for the `chrono` based timestamp type used in the database.
+pub type DbTimestamp = chrono::DateTime<chrono::Utc>;
+
+/// Represents an already opened transaction.
+/// This is a thin wrapper over [DatabaseTransaction].
+/// The wrapping behavior is needed to allow transaction agnostic functionalities
+/// of the DB traits.
+#[derive(Debug)]
+pub struct OpenTransaction(DatabaseTransaction);
+
+impl OpenTransaction {
```

Comment:
Could be loaded from the `db/api` crate and redeclared by this crate to avoid code duplicity.